src/components/AugmentedResultList/AugmentedResultList.ts

```diff
@@ -0,0 +1,142 @@
+            return null;
+        }),
+    };
+
+    /**
+     * Creates a new `AugmentedResultList` component.
+     * @param element The HTMLElement on which to instantiate the component.
+     * @param options The options for the `ResultList` component.
+     * @param bindings The bindings that the component requires to function normally. If not set, these will be
+     * automatically resolved (with a slower execution time).
+     */
+    constructor(public element: HTMLElement, public options: AugmentedResultListOptions, public bindings: IComponentBindings) {
+        super(element, ComponentOptions.initComponentOptions(element, AugmentedResultList, options), bindings, AugmentedResultList.ID);
+    }
+
+    private getObjectPayload(results: IQueryResult[]): String[] {
+        const field = this.getFieldString(this.options.matchingIdField);
+        if (results.length > 0) {
+            return results.filter((result) => result.raw && result.raw[field]).map((result) => result.raw[field]);
+        }
+        return [];
+    }
+
+    private getFieldString(fieldName: IFieldOption) {
+        return fieldName.replace('@', '');
+    }
+
+    protected enableAnimation() {
+        if (!document.getElementById('overlay')) {
+            $$(document.body).append($$('div', { id: 'overlay', class: 'modal-backdrop fade in' }).el);
+        }
+    }
+
+    protected disableAnimation() {
+        if (document.getElementById('overlay')) {
+            document.getElementById('overlay').remove();
+        }
+    }
+
+    public renderResults(resultElements: HTMLElement[], append = false): Promise<void> {
+        const res = super.renderResults(resultElements, append);
+        this.disableAnimation();
+        return res;
+    }
+
+    public async buildResults(results: Coveo.IQueryResults): Promise<HTMLElement[]> {
+        this.enableAnimation();
+        const fieldString = this.getFieldString(this.options.matchingIdField);
+
+        if (this.options.objectDataAction) {
+            // Call remote action to fetch object data
+            const remoteResults: IPromiseReturnArgs = await this.options
+                .objectDataAction(this.getObjectPayload(results.results))
+                .then((data) => {
+                    return data;
+                })
+                .catch((ex) => {
+                    this.logger.error('Unable to fetch object data.');
+                    return null;
+                });
+
+            if (remoteResults && remoteResults.data) {
+                // Merge remote action results with Coveo Results
+                results.results.forEach((res: Coveo.IQueryResult) => {
+                    const match = remoteResults.data.objectData.find((data) => {
+                        return (data as any)[fieldString] === res.raw[fieldString];
+                    });
+
+                    // Attach data specific to each result/object
+                    for (const key in match) {
```

Comment:
Can you change the for loops for their functionnal variant ? (Object.keys().forEach)

Reply:
I wouldn't mind doing it but what's the problem you see with this?

Reply:
I gets a bit messier tbh...
`remoteResults.data.commonData.keys().forEach((key: any) => {`
instead of
`for (const key in remoteResults.data.commonData) {`

Reply:
the for in loop had some issue in IE 11 iirc.

Reply:
@jeremierobert-coveo you're probably mixing up the for..of and the for..in:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in

Reply:
correct. it's for...of that causes issues.

Reply:
I want to leave it as is